charmpath.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
+
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+)
+
+// CharmPath is used to access a charm at a given path.
+type CharmPath interface {
+	// Charm returns the charm represented by this path.
+	// If the series is not specified, the charm's default
+	// series is used, if any. Otherwise, the series is
+	// validated against those the charm declares it supports.
+	Charm(series string) (charm.Charm, *charm.URL, error)
+}
+
+// charmPath represents a path (eg local directory) containing a
+// single charm.
+type charmPath struct {
+	Name      string
+	CharmInfo charm.Charm
+}
+
+var _ CharmPath = (*charmPath)(nil)
+
+// NewCharmPath creates and returns an instance used to
+// open a charm located at the given path.
+func NewCharmPath(path string) (CharmPath, error) {
```

Comment:
I'm not convinced that we need the Charm(series) method.
The charm.SeriesForCharm function gets us almost all of the
way there anyway.

Assuming NewCharmPath returns (charm.Charm, *charm.Reference, error),
a client can do:

    c, url, err := charmrepo.NewCharmPath(path)
    if err != nil {...}
    url.Series, err = charm.SeriesForCharm(requestedSeries, c.Meta().Series)

to resolve the series.
I'd suggest that SeriesForCharm returns an error if requestSeries=="" and
there are no series declared in the charm metadata.

I still think that the CharmPath interface is unnecessary.